metrics/metrics-dropwizard/src/test/java/org/apache/james/metrics/dropwizard/DropWizardMetricFactoryTest.java

```diff
@@ -38,4 +46,26 @@ void setUp() {
     public MetricFactory testee() {
         return testee;
     }
+
+    @Test
+    void decoratePublisherWithTimerMetricShouldRecordANewValueForEachRetry() {
```

Comment:
can we, with this API, measure the latency of a service "from the outside"?
Let's say we want to know what is the latency of getting a blob from S3, I don't care if when try 3 times the third one is fast if from a consumer point of view the addition of these latencies makes a big amount of time.
I understand that in some cases we want a instance of measure for each retry but both cases make sense, right?

Reply:
Well, let's take an exemple: we retry LocalDelivery.

What you describe is represented by the metric measuring the service latency at the level of LocalDelivery. Which is different from "how long does it takes to save a blob". (I would agree with you if the retry is performed as part of the blobStore component).

Also tests highlighted you could "trick" with external operations could get recorded as well.

```
decoratePublisherWithTimerMetric("plop", mono)
   .delay(100 ms) // perform some other unrelated operations like indexing data
   .retry(5) 
```

Would record the unrelated operation as part of reading blob (!) which leads to relevant metrics, and lead to false diagnostic.
```

Reply:
Sorry, I can't understand your message despite reading it 3 times. You seems to disagree, take an example but I can't understand if we can measure both timings or not (whatever the API we use) and what you disagree with.

Reply:
I don't know how to explain it better.

Should we plan a pair on exploring this together, so that you help me better formulating my answer?

Reply:
maybe someone else can try to explain? I see that @Arsnael approved the PR so he may explain with different words that I could understand? Or maybe @remk @blackheaven or @rouazana could try to read your explanation and if they understand they can explain it back to me?

Reply:
maybe we could have something like : 

```
decoratePublisherWithTimerMetric("plopTotalTime",
decoratePublisherWithTimerMetric("plop", mono)
   .delay(100 ms) // perform some other unrelated operations like indexing data
   .retry(5)
)
```

Reply:
Yes, if that snippet works, It's perfectly ok

Reply:
it will be needed to update the grafana boards to include both metrics.

Reply:
Well the problem is, with master code, "plop" metric gonna be recording a single value, being `"plop" + TOTAL_TIME_SUFFIX`, that's what we tests and fix here.